internal/cmd/server.go

```diff
@@ -21,7 +25,16 @@ func newServerCmd() *cobra.Command {
 		RunE: func(cmd *cobra.Command, _ []string) error {
 			logging.SetServerLogger()
 
-			if err := parseOptions(cmd, &options, "INFRA_SERVER"); err != nil {
+			if configFilename == "" {
+				configFilename = os.Getenv("INFRA_SERVER_CONFIG_FILE")
+			}
```

Comment:
This, and the binding of the value to this variable below, are necessary because the config filename needs to be parsed first before anything else, but all other flags and env vars are used as overrides to the values in the config file.

So the path to a config file will always be a special case, but I think that is expected given that it doesn't follow the regular precedence rules, and because we don't pass the config filename in to `server.New`.